tests/lib/analyze-features.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+before_non_nested_task() {
+    # Add a recognizable unique string to the journal logs.
+    # This will be the marker for the journal-analyzer.py script to start parsing the logs.
+    echo "end-of-prepare-suite-each for $SPREAD_JOB" | systemd-cat
+}
+
+before_nested_task() {
+    # Add a recognizable unique string to the journal logs.
+    # This will be the marker for the journal-analyzer.py script to start parsing the logs.
+    "$TESTSTOOLS"/remote.exec "echo \"end-of-prepare-suite-each for $SPREAD_JOB\" | systemd-cat"
+}
+
+after_non_nested_task() {
+    # Write to the directory specified in the spread.yaml file for artifacts
+    local write_dir="${SPREAD_PATH}/feature-tags"
+    mkdir -p "$write_dir"
+    "$TESTSTOOLS"/journal-analyzer.py -t "end-of-prepare-suite-each for $SPREAD_JOB" -f "$TAG_FEATURES" -o "${write_dir}/${SPREAD_JOB//\//\\}"
```

Comment:
what if we encode `/` as `--` or `__` which are unlikely to show up in the test names ? this would avoid awkward looking names and needing to be careful about quoting and escaping
```suggestion
    "$TESTSTOOLS"/journal-analyzer.py -t "end-of-prepare-suite-each for $SPREAD_JOB" -f "$TAG_FEATURES" -o "${write_dir}/${SPREAD_JOB//\//--}"
```

Reply:
That's a great suggestion. I'll do that.